packages/pug-code-gen/index.js

```diff
@@ -97,15 +110,64 @@ Compiler.prototype = {
     throw err;
   },
 
+  parseExpr: function(expr) {
+    //return babylon.parse('function*g(){return e='+expr+'}').program.body[0].body.body[0].argument.right;
+    return babylon.parseExpression(expr);
+  },
+  parseArgs: function(args) {
+    return babylon.parse("a("+args+")").program.body.pop().expression.arguments;
+  },
+  btpl_addWith: function() {
+    var globals = this.options.globals ? this.options.globals.concat(INTERNAL_VARIABLES) : INTERNAL_VARIABLES;
+    globals.concat(this.runtimeFunctionsUsed.map(function (name) { return 'pug_' + name; }));
+    var tpl = "// @with exclude: "+ globals.join(",") +"\n{\nlocals || {};\nSOURCE;\n}"
+    var tplc = babelTemplate(tpl, { preserveComments: true });
+    return tplc;
+  },
+  ast_variableDeclaration: function() {
+    return t.variableDeclaration('var', [
+          t.variableDeclarator(t.identifier('pug_html'), t.stringLiteral('')),
+          t.variableDeclarator(t.identifier('pug_mixins'), t.objectExpression([])),
+          t.variableDeclarator(t.identifier('pug_interp'), null)
+        ])
+  },
+  ast_return: function() {
+    return [t.emptyStatement(), t.returnStatement(t.identifier('pug_html'))];
+  },
+  ast_stringify: function(lit) {
+    lit.extra = { rawValue: lit.value, raw: stringify(lit.value) };
+    return lit;
+  },
+  wrapCallExpression: function(node) {
```

Comment:
Is this intended to act as a plugin hook at some point?

Reply:
sort of. This is one of the main inheritance point for then-pug

```javascript
Compiler.prototype.wrapCallExpression = function(node) {
  return t.yieldExpression(node, true);
}
```